File: app/src/main/java/com/veganbeauty/app/features/shop/home/ShopHomeBannerAdapter.java
package com.veganbeauty.app.features.shop.home;

import android.view.LayoutInflater;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.DiffUtil;
import androidx.recyclerview.widget.ListAdapter;
import androidx.recyclerview.widget.RecyclerView;

import com.veganbeauty.app.databinding.ShopHomeBannerBinding;
import com.veganbeauty.app.features.shop.home.models.BannerUiModel;

public class ShopHomeBannerAdapter extends ListAdapter<BannerUiModel, ShopHomeBannerAdapter.BannerViewHolder> {

    public ShopHomeBannerAdapter() {
        super(new DiffCallback());
    }

    @NonNull
    @Override
    public BannerViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        ShopHomeBannerBinding binding = ShopHomeBannerBinding.inflate(LayoutInflater.from(parent.getContext()), parent, false);
        return new BannerViewHolder(binding);
    }

    @Override
    public void onBindViewHolder(@NonNull BannerViewHolder holder, int position) {
        holder.bind(getItem(position));
    }

    public static class BannerViewHolder extends RecyclerView.ViewHolder {
        private final ShopHomeBannerBinding binding;

        public BannerViewHolder(@NonNull ShopHomeBannerBinding binding) {
            super(binding.getRoot());
            this.binding = binding;
        }

        public void bind(BannerUiModel banner) {
            binding.ivBanner.setImageResource(banner.getImageRes());
        }
    }

    public static class DiffCallback extends DiffUtil.ItemCallback<BannerUiModel> {
        @Override
        public boolean areItemsTheSame(@NonNull BannerUiModel oldItem, @NonNull BannerUiModel newItem) {
            return oldItem.getId().equals(newItem.getId());
        }

        @Override
        public boolean areContentsTheSame(@NonNull BannerUiModel oldItem, @NonNull BannerUiModel newItem) {
            return oldItem.equals(newItem);
        }
    }
}

File: app/src/main/java/com/veganbeauty/app/features/shop/home/ShopHomeBannerAdapter.kt
package com.veganbeauty.app.features.shop.home

import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.recyclerview.widget.DiffUtil
import androidx.recyclerview.widget.ListAdapter
import androidx.recyclerview.widget.RecyclerView
import com.veganbeauty.app.databinding.ShopHomeBannerBinding
import com.veganbeauty.app.features.shop.home.models.BannerUiModel

class ShopHomeBannerAdapter : ListAdapter<BannerUiModel, ShopHomeBannerAdapter.BannerViewHolder>(DiffCallback()) {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): BannerViewHolder {
        val binding = ShopHomeBannerBinding.inflate(LayoutInflater.from(parent.context), parent, false)
        return BannerViewHolder(binding)
    }

    override fun onBindViewHolder(holder: BannerViewHolder, position: Int) {
        holder.bind(getItem(position))
    }

    class BannerViewHolder(private val binding: ShopHomeBannerBinding) : RecyclerView.ViewHolder(binding.root) {
        fun bind(banner: BannerUiModel) {
            // Tạm thời hiển thị màu nền hoặc ảnh mặc định
            // Sau này bạn có thể dùng Coil/Glide: binding.ivBanner.load(banner.imageUrl)
            binding.ivBanner.setImageResource(banner.imageRes)
        }
    }

    class DiffCallback : DiffUtil.ItemCallback<BannerUiModel>() {
        override fun areItemsTheSame(oldItem: BannerUiModel, newItem: BannerUiModel) = oldItem.id == newItem.id
        override fun areContentsTheSame(oldItem: BannerUiModel, newItem: BannerUiModel) = oldItem == newItem
    }
}
